tests/helper.go

```diff
@@ -401,6 +405,8 @@ func TestEvalExprModifiedOrigin(t *testing.T, tt *EvalTestItem, from, until int6
 			t.Errorf("metric[%d] %+v with name %v doesn't contain 'name' tag", i, actual, actual.Name)
 		}
 		for k, v := range want.Tags {
+			// FIXME: we are skipping Tags comparison until we fix the tests https://github.com/grafana/carbonapi/issues/65
```

Comment:
Need to be fixed. It will mask overwrite tags without copy-on-write. And produce some bugs. Lookup the failed tests of functions and and copy-onwrite when needed.

Reply:
So the PR they are refering to here says:
```
Solving this in the upstream sync PR would have made the PR huge. So I'm creating this issue to solve it separately, and we will be skipping the Tags comparison for now.
```

I would say this is for now valid, and can indeed be fixed in a separate PR later (this is already a big set of change and I would agree that fixing that in the same PR would make it much harder to review).

Reply:
> I would say this is for now valid, and can indeed be fixed in a separate PR later (this is already a big set of change and I would agree that fixing that in the same PR would make it much harder to review).
May be, but need don't forget about this. No-Copy-On-Write bugs is my nightmare :-) I spent quite a lot of time fixing a some  of bugs in huge functions pipeline.

Reply:
We won't forget. But to clarify: this is not about copy-on-write, the assertions on the original data being unmodified are still there (and tests are passing).

This is commenting out the assertions on the **tags of the output**. And we are commenting them out not because the functions implementations are wrong, but because the tests are not currently building the "expected" MetricData correctly.

Reply:
Yes, but some protection and test correction is needed. But may be not in this PR.

Reply:
Will send this in another PR once this is merged. I'll resolve this now, feel free to re-open if you disagree.